exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DescribeSchemaHandler.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.drill.exec.planner.sql.SchemaUtilites;
+import org.apache.drill.exec.planner.sql.parser.SqlDescribeSchema;
+import org.apache.drill.exec.store.StoragePlugin;
+import org.apache.drill.exec.store.dfs.FileSystemSchemaFactory;
+import org.apache.drill.exec.store.dfs.WorkspaceConfig;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.fasterxml.jackson.databind.SerializationFeature.INDENT_OUTPUT;
+
+public class DescribeSchemaHandler extends DefaultSqlHandler {
+
+  public DescribeSchemaHandler(SqlHandlerConfig config) {
+    super(config);
+  }
+
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DescribeSchemaHandler.class);
+  private static final ObjectMapper mapper = new ObjectMapper(new ObjectMapper().getFactory().setCharacterEscapes(new CharacterEscapes() {
+    @Override
+    public int[] getEscapeCodesForAscii() {
+      // add standard set of escaping characters
+      int[] esc = CharacterEscapes.standardAsciiEscapesForJSON();
+      // don't escape backslash (not to corrupt windows path)
+      esc['\\'] = CharacterEscapes.ESCAPE_NONE;
+      return esc;
+    }
+
+    @Override
+    public SerializableString getEscapeSequence(int i) {
+      // no further escaping (beyond ASCII chars) needed
+      return null;
+    }
+  })).enable(INDENT_OUTPUT);
+
+
+  @Override
+  public PhysicalPlan getPlan(SqlNode sqlNode) {
+    SqlIdentifier schema = ((SqlDescribeSchema) sqlNode).getSchema();
+    SchemaPlus drillSchema = SchemaUtilites.findSchema(config.getConverter().getDefaultSchema(), schema.names);
+
+    if (drillSchema != null) {
+      StoragePlugin storagePlugin;
+      try {
+        storagePlugin = context.getStorage().getPlugin(schema.names.get(0));
+      } catch (ExecutionSetupException e) {
+        throw UserException.validationError()
+            .message("Failure while retrieving storage plugin", e)
+            .build(logger);
+      }
+      String properties;
+      try {
+        properties = getPropertiesAsJsonString(schema.names, storagePlugin.getConfig());
+      } catch (JsonProcessingException e) {
+        throw UserException.parseError()
+            .message("Error while trying to convert storage config to json string")
+            .build(logger);
+      }
+      return DirectPlan.createDirectPlan(context, new DescribeSchemaResult(Joiner.on(".").join(schema.names), properties));
+    }
+
+    throw UserException.validationError()
+          .message(String.format("Invalid schema name [%s]", Joiner.on(".").join(schema.names)))
+          .build(logger);
+  }
+
+  /**
+   * Converts schema config properties to json string. If storage plugin has several workspaces, picks appropriate one.
+   */
+  private String getPropertiesAsJsonString(List<String> names, StoragePluginConfig config) throws JsonProcessingException {
+    final Map configMap = mapper.convertValue(config, Map.class);
+    Object workspaces = configMap.remove("workspaces");
```

Comment:
The logic of L104 - 117 seems to apply to FileSystemSP only. Better make sure it will be called only for FilesystemSP (what if other storage plugin config also adds "workspace" later on? )